### PR TITLE
GT-1498 All Tools Spotlight

### PR DESCRIPTION
**Pull request by rachaelblue**
https://jira.cru.org/browse/GT-1498

### Things I Did
- refactored the ToolCardView-- broke it out into several subviews to make it easier to read
- adjusted ToolCardView to resize when used in the Spotlight section
- added a Realm attribute for the spotlight on Resource
- added strings to the base localization files (and sent the strings/ids to Ann to add translations)
- refactored AllToolsList so that we don't have separate versions for iOS 13/14/15.  Now, the only version check happens in AllToolsContentView, and all versions use the same AllToolsList (@levieggertcru, I realized this would work better than what we previously discussed 😄)

### CODE REVIEW

**Review by levieggertcru — CHANGES_REQUESTED** [review]
Hey @rachaelblue this is looking great.  Here's a recap of those changes we discussed:
- Ensure AppDiContainer dependencies are injected into Mock viewModels.
Also, created this task here for creating the Mock swift local package for all of our Mock files https://jira.cru.org/browse/GT-1577.

File: godtools/App/Features/Tools/SwiftUI/ToolCardView/ViewModels/MockToolCardViewModel.swift [hunk 1]
```diff
@@ -21,8 +20,8 @@ class MockToolCardViewModel: BaseToolCardViewModel {
             parallelLanguageName = "Arabic (Bahrain) ✓"
         }
         
-        if showBannerImage, let deviceImage = appDiContainer.deviceAttachmentBanners.getDeviceBanner(resourceId: "2") {
-            bannerImage = Image(uiImage: deviceImage)
+        if showBannerImage {
+            bannerImage = Image.from(uiImage: DeviceAttachmentBanners().getDeviceBanner(resourceId: "2"))
```
Comment (levieggertcru) [inline godtools/App/Features/Tools/SwiftUI/ToolCardView/ViewModels/MockToolCardViewModel.swift:24]: Hey @rachaelblue can you actually inject the DeviceAttachmentBanners into the ViewModel and fetch it from the SwiftUIPreviews AppDiContainer.  I should have been more clear on that.  Eventually the MockDiContainer will provide an instance that can be injected.

File: godtools/App/Features/Tools/SwiftUI/ToolCardView/SubViews/ToolCardBannerImageView.swift [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            .cornerRadius(cornerRadius, corners: [.topLeft, .topRight])
+    }
+}
+
+// MARK: - Preview
+
+struct ToolCardBannerImageView_Previews: PreviewProvider {
+    static var previews: some View {
+        let isSpotlight = true
+        let bannerImage = DeviceAttachmentBanners().getDeviceBanner(resourceId: "2")
```
Comment (levieggertcru) [inline godtools/App/Features/Tools/SwiftUI/ToolCardView/SubViews/ToolCardBannerImageView.swift:50]: Same here can you fetch the DeviceAttachmentBanners from the SwiftUIPreviews AppDiContainer.  I wasn't clear on that.

**Review by levieggertcru — CHANGES_REQUESTED** [review]

File: godtools/App/Features/Tools/SwiftUI/ToolCardView/ViewModels/MockToolCardViewModel.swift (hunk 1 above)
Comment (rachaelblue) [reply godtools/App/Features/Tools/SwiftUI/ToolCardView/ViewModels/MockToolCardViewModel.swift]: That makes sense.  Fixed!

**Review by levieggertcru — APPROVED** [review]

**Pull request merged**